src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -98,37 +99,34 @@ def llm_classify(
 
     eval_template = normalize_template(template)
     prompts = map_template(dataframe, eval_template)
-    with set_verbosity(model, verbose) as verbose_model:
-        responses = verbose_model.generate(
-            prompts.to_list(), instruction=system_instruction, **model_kwargs
-        )
-
     labels: List[str] = []
     explanations: List[Optional[str]] = []
+    if generation_info := model.verbose_generation_info():
+        printif(verbose, generation_info)
 
-    printif(verbose, f"Snapping {len(responses)} responses to rails: {rails}")
-    for response in responses:
-        if not use_openai_function_call:
-            raw_string = response
-            if provide_explanation:
-                # TODO: support explanation without function calling
-                explanations.append(None)
-        else:
-            try:
-                function_arguments = json.loads(response, strict=False)
-                raw_string = function_arguments.get(_RESPONSE)
-                if provide_explanation:
-                    explanations.append(function_arguments.get(_EXPLANATION))
-            except json.JSONDecodeError:
-                raw_string = response
-        labels.append(_snap_to_rail(raw_string, rails, verbose=verbose))
-
-    data: Dict[str, List[Any]] = {"label": labels}
-    if provide_explanation:
-        assert len(labels) == len(explanations)
-        data["explanation"] = explanations
-
-    return pd.DataFrame(data=data, index=dataframe.index)
+    with set_verbosity(model, verbose) as verbose_model:
+        for prompt in tqdm(prompts):
+            response = verbose_model(prompt, instruction=system_instruction, **model_kwargs)
+            if not use_openai_function_call:
+                unrailed_label = response
+                explanation = None
+            else:
+                try:
+                    function_arguments = json.loads(response, strict=False)
+                    unrailed_label = function_arguments.get(_RESPONSE)
+                    explanation = function_arguments.get(_EXPLANATION)
+                except json.JSONDecodeError:
+                    unrailed_label = response
+                    explanation = None
+            labels.append(_snap_to_rail(unrailed_label, rails, verbose=verbose))
+            explanations.append(explanation)
+        return pd.DataFrame(
```

Comment:
nit. This part doesn't need to be indented. Alternatively we can also just nest just the model part so we don't have to indent everything.

Reply:
addressed in [9ee08c5](https://github.com/Arize-ai/phoenix/pull/1712/commits/9ee08c5a7d83f85fea4a9731d9768e7ea4639703)